pkg/virt-controller/watch/vmi.go

```diff
@@ -380,6 +380,28 @@ func (c *VMIController) updateStatus(vmi *virtv1.VirtualMachineInstance, pod *k8
 			conditionManager.RemoveCondition(vmiCopy, virtv1.VirtualMachineInstanceConditionType(k8sv1.PodReady))
 		}
 
+		// IF the VMI has pod network, Check and patch if IP on VMI is not same as the one on Pod IP
+		for _, network := range vmiCopy.Spec.Networks {
+			if network.NetworkSource.Pod != nil {
+				for i, iface := range vmiCopy.Status.Interfaces {
+					if iface.Name == network.Name {
+						vmiCopy.Status.Interfaces[i].IP = pod.Status.PodIP
```

Comment:
This  looks good, but we also have an IPs field. You will also have to search in the list there if it is present.
Further I think that virt-handler will always override the IP with the link-local address of the VM which is reported by the qemu guest agent. We will also have to update the status update in virt-handler so that it respects the pod IP if `masquerade` is set and only addes the link-local address to `IPs`.

Reply:
The IPs field is handled on the line 390, is not it? Maybe it has been added since the comment.

We first report the pod interface with the pod IP here: https://github.com/kubevirt/kubevirt/blob/master/pkg/virt-controller/watch/vmi.go#L339, no guest agent at this point.

The guest agent reporting is handled here https://github.com/kubevirt/kubevirt/blob/master/pkg/virt-handler/vm.go#L275. And I believe it overwrites the orignal Pod interface report here https://github.com/kubevirt/kubevirt/blob/master/pkg/virt-handler/vm.go#L315 in case the guest agent is running. It smells though, I'd expect we have that tested, it should still report the public pod IP, not the link local.

Reply:
@vatsalparekh do you have a KubeVirt cluster running? Could you check what IP is reported on a VMI with masquerade when guest agent is installed?

Reply:
@petrkotas 
```
➜  kubevirt git:(migration-bindings-fix) ✗ kg vmi -o wide
NAME         AGE   PHASE     IP            NODENAME   LIVE-MIGRATABLE
vmi-fedora   22m   Running   10.0.2.2/24   node01     True
```
```
    interfaces:
    - interfaceName: eth0
      ipAddress: 10.0.2.2/24
      ipAddresses:
      - 10.0.2.2/24
      - fe80::ff:fe72:7c3c/64
      mac: 02:00:00:72:7c:3c
      name: default

```

Reply:
Thanks. It is an unrelated bug. Let's tackle it in a separate PR.